server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheTests.java

```diff
@@ -198,13 +203,93 @@ public void testCacheDifferentReaders() throws Exception {
         assertEquals(0, requestCacheStats.stats().getEvictions());
         assertTrue(loader.loadedFromCache);
         assertEquals(0, cache.count());
-        assertEquals(0, requestCacheStats.stats().getMemorySize().bytesAsInt());
+        assertEquals(0L, requestCacheStats.stats().getMemorySize().getBytes());
 
         IOUtils.close(secondReader, writer, dir, cache);
         assertEquals(0, cache.numRegisteredCloseListeners());
     }
 
+    public void testCacheDifferentMapping() throws Exception {
+        IndicesRequestCache cache = new IndicesRequestCache(Settings.EMPTY);
+        MappingLookup.CacheKey mappingKey1 = MappingLookupUtils.fromTypes().cacheKey();
+        MappingLookup.CacheKey mappingKey2 = MappingLookupUtils.fromTypes().cacheKey();
```

Comment:
could we simply use an empty mapping lookup here instead of calling fromTypes without providing any type?

Reply:
We could use it for one of the keys I guess but I wanted distinct cache keys for both keys and this manages it.

Reply:
oh boy you are right.